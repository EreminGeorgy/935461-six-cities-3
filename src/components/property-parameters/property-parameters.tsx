import * as React from "react";
import {useCallback} from "react";
import {getStars} from "../../utils/utils";
import {Review} from "../review/review";
import ReviewSend from "../review-send/review-send";
import {CityMap} from "../city-map/city-map";
import {Operation} from "../../reducer/favorites/favorites";
import {CommentData, Offer} from "../../types";
import {connect} from "react-redux";

const COMMENTS_TO_SHOW = 10;

interface Props {
  offer: Offer;
  closestOffers: Offer[];
  path: string;
  activeCard: number;
  comments: CommentData[];
  updateComments: (id: number) => void;
  changeCard: ({id, status}: {id: number; status: number}) => void;
}

export const PropertyParameters: React.FunctionComponent<Props> = (props: Props) => {
  const {offer, closestOffers, path, activeCard, comments, updateComments, changeCard} = props;

  const {
    id,
    imagesSrc,
    title,
    description,
    price,
    rating,
    type,
    isPremium,
    isFavorite,
    bedrooms,
    guests,
    householdItems,
    host,
    city,
  } = offer;

  const handleFavoriteClick = (evt) => {

    evt.preventDefault();

    changeCard({
      id,
      status: Number(!isFavorite),
    });
  };

  const memoClick = useCallback(handleFavoriteClick, [isFavorite]);

  const width = getStars(rating);
  const availableItems = Array.from(householdItems);

  const activeComments = comments.sort((a: any, b: any) => {
    return parseInt(a.dateString, 10) - parseInt(b.dateString, 10);
  }).slice(-(COMMENTS_TO_SHOW));

  return (
    <section className="property">
      <div className="property__gallery-container container">
        <div className="property__gallery">
          {imagesSrc.map((src, i) => {
            return (<div className="property__image-wrapper" key={`src-${i}`}>
              <img className="property__image" src={src} alt="Photo studio"/>
            </div>);
          })}
        </div>
      </div>
      <div className="property__container container">
        <div className="property__wrapper">
          {isPremium ? (<div className="property__mark"><span>Premium</span></div>) : (``)}
          <div className="property__name-wrapper">
            <h1 className="property__name">{title}</h1>
            <button className={`property__bookmark-button button ${isFavorite ? `property__bookmark-button--active` : ``}`}
              onClick={memoClick} type="button">
              <svg className="place-card__bookmark-icon" width="31" height="33">
                <use xlinkHref="#icon-bookmark"></use>
              </svg>
              <span className="visually-hidden">To bookmarks</span>
            </button>
          </div>
          <div className="property__rating rating">
            <div className="property__stars rating__stars">
              <span style={{width}}></span>
              <span className="visually-hidden">Rating</span>
            </div>
            <span className="property__rating-value rating__value">{rating}</span>
          </div>
          <ul className="property__features">
            <li className="property__feature property__feature--entire">
              {type}
            </li>
            <li className="property__feature property__feature--bedrooms">
              {bedrooms} Bedrooms
            </li>
            <li className="property__feature property__feature--adults">
              Max {guests} adults
            </li>
          </ul>
          <div className="property__price">
            <b className="property__price-value">&euro;{price}</b>
            <span className="property__price-text">&nbsp;night</span>
          </div>
          <div className="property__inside">
            <h2 className="property__inside-title">What&apos;s inside</h2>
            <ul className="property__inside-list">
              {availableItems.map((item) => {
                return (<li className="property__inside-item" key={item.toString()}>
                  {item}
                </li>);
              })}
            </ul>
          </div>April 2019
          <div className="property__host">
            <h2 className="property__host-title">Meet the host</h2>
            <div className="property__host-user user">
              <div className={`property__avatar-wrapper ${host.isPro ? `property__avatar-wrapper--pro` : ``} user__avatar-wrapper`}>
                <img className="property__avatar user__avatar" src={host.avatarUrl} width="74" height="74" alt="Host avatar"/>
              </div>
              <span className="property__user-name">
                {host.name}
              </span>
            </div>
            <div className="property__description">
              <p className="property__text" >{description}</p>
            </div>
          </div>
          <section className="property__reviews reviews">
            <h2 className="reviews__title">Reviews &middot; <span className="reviews__amount">{comments.length}</span></h2>
            <ul className="reviews__list">
              {activeComments.map((commentData) => {
                return <Review
                  key={commentData.id}
                  commentData={commentData}
                />;
              })}
            </ul>
            <ReviewSend
              id={offer.id}
              updateComments={updateComments}
            />
          </section>
        </div>
      </div>
      <CityMap
        offers={closestOffers}
        city={city}
        path={path}
        activeCard={activeCard}
        currentOfferCoords={offer.locations}
      />
    </section>
  );
};


const mapDispatchToProps = (dispatch) => ({
  changeCard(cardData) {
    dispatch(Operation.changeStatus(cardData));
  },
});

export default connect(null, mapDispatchToProps)(PropertyParameters);
