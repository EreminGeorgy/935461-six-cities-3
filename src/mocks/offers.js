const shortid = require(`shortid`);

// const MOCKS_TOTAL = 10;
// const IMAGES = [
//   `img/apartment-01.jpg`,
//   `img/apartment-02.jpg`,
//   `img/apartment-01.jpg`,
//   `img/apartment-03.jpg`,
//   `img/apartment-02.jpg`,
//   `img/room.jpg`
// ];
// const TITLES = [
//   `Wood and stone place`,
//   `Beautiful &amp; luxurious apartment at great location`,
//   `Canal View Prinsengracht`,
//   `Nice, cozy, warm big bed apartment`,
// ];
// const TYPES = [
//   `Apartment`,
//   `Private room`,
//   `house`,
//   `hotel`,
// ];
// const HOUSEHOLD_ITEMS = [
//   `Wi-Fi`,
//   `Washing machine`,
//   `Towels`,
//   `Heating`,
//   `Coffee machine`,
//   `Baby seat`,
//   `Kitchen`,
//   `Cabel TV`,
//   `Fridge`,
// ];
// const USER_NAMES = [
//   `Angelina`,
//   `Max`,
// ];
// const USER_AVATAR_SRC = [
//   `img/avatar-angelina.jpg`,
//   `img/avatar-max.jpg`,
// ];
// const DESCRIPTION_TEXT = [
//   `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
//   `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
// ];

// const randomBoolean = () => Boolean(Math.round(Math.random()));
// const randomNumber = ({min = 0, max = 1}) => {
//   return min + Math.floor(max * Math.random());
// };

// const getMock = () => ({
//   id: shortid.generate(),
//   city: citiesData[Math.floor(Math.random() * 5)],
//   previewSrc: IMAGES[Math.floor(Math.random() * 6)],
//   imagesSrc: IMAGES,
//   title: TITLES[Math.floor(Math.random() * 4)],
//   description: DESCRIPTION_TEXT,
//   isPremium: randomBoolean(),
//   type: TYPES[Math.floor(Math.random() * 4)],
//   raiting: randomNumber({min: 0, max: 5}),
//   bedrooms: randomNumber({min: 1, max: 3}),
//   guests: randomNumber({min: 1, max: 9}),
//   price: randomNumber({min: 0, max: 999}),
//   householdItems: new Set(HOUSEHOLD_ITEMS),
//   host: {
//     name: USER_NAMES[Math.floor(Math.random() * 2)],
//     isPro: randomBoolean(),
//     avatarUrl: USER_AVATAR_SRC[Math.floor(Math.random() * 2)],
//   },
//   location: [
//     [52.370955, 4.873096],
//     [52.379553, 4.873096],
//     [52.390955, 4.909309],
//     [52.360955, 4.909309]
//   ]
// });

// const getOffers = () => {
//   return new Array(MOCKS_TOTAL).fill(``).map(getMock);
// }

// const offers = getOffers();

const offers = [
  {
    id: shortid.generate(),
    city: {
      name: `Cologne`,
      location: [50.939485, 6.948775],
    },
    imagesSrc: [
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-01.jpg`,
      `img/apartment-03.jpg`,
      `img/apartment-02.jpg`,
      `img/room.jpg`
    ],
    previewSrc: `img/apartment-01.jpg`,
    title: `_`,
    price: 0,
    raiting: 0,
    type: `Apartment`,
    isPremium: true,
    bedrooms: 3,
    guests: 2,
    householdItems: new Set([
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
    ]),
    host: {
      name: `Angelina`,
      isPro: true,
      avatarUrl: `img/avatar-angelina.jpg`,
    },
    location: [50.937764, 6.961390],
  },
  {
    id: shortid.generate(),
    city: {
      name: `Paris`,
      location: [48.859335, 2.350730],
    },
    imagesSrc: [
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-01.jpg`,
      `img/apartment-03.jpg`,
      `img/apartment-02.jpg`,
      `img/room.jpg`
    ],
    previewSrc: `img/apartment-02.jpg`,
    title: `Wood and stone place`,
    price: 90,
    raiting: 8.35,
    type: `Private room`,
    isPremium: false,
    bedrooms: 3,
    guests: 2,
    householdItems: new Set([
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
    ]),
    host: {
      name: `Angelina`,
      isPro: true,
      avatarUrl: `img/avatar-angelina.jpg`,
    },
    location: [48.872344, 2.360418],
  },
  {
    id: shortid.generate(),
    city: {
      name: `Brussels`,
      location: [50.855889, 4.351608],
    },
    imagesSrc: [
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-01.jpg`,
      `img/apartment-03.jpg`,
      `img/apartment-02.jpg`,
      `img/room.jpg`
    ],
    previewSrc: `img/apartment-03.jpg`,
    title: `Canal View Prinsengracht`,
    price: 80,
    raiting: 8.2,
    type: `1`,
    isPremium: true,
    bedrooms: 3,
    guests: 2,
    householdItems: new Set([
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
    ]),
    host: {
      name: `Angelina`,
      isPro: true,
      avatarUrl: `img/avatar-angelina.jpg`,
    },
    location: [50.855889, 4.351608]
  },
  {
    id: shortid.generate(),
    city: {
      name: `Amsterdam`,
      location: [52.371512, 4.893750],
    },
    imagesSrc: [
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-01.jpg`,
      `img/apartment-03.jpg`,
      `img/apartment-02.jpg`,
      `img/room.jpg`
    ],
    previewSrc: `img/room.jpg`,
    title: `Nice, cozy, warm big bed apartment`,
    price: 50,
    raiting: 3,
    type: `Apartment`,
    isPremium: false,
    bedrooms: 3,
    guests: 2,
    householdItems: new Set([
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
    ]),
    host: {
      name: `Angelina`,
      isPro: true,
      avatarUrl: `img/avatar-angelina.jpg`,
    },
    location: [52.370955, 4.873096]
  },
  {
    id: shortid.generate(),
    city: {
      name: `Amsterdam`,
      location: [52.371512, 4.893750],
    },
    imagesSrc: [
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-01.jpg`,
      `img/apartment-03.jpg`,
      `img/apartment-02.jpg`,
      `img/room.jpg`
    ],
    previewSrc: `img/room.jpg`,
    title: `Nice, cozy, warm big bed apartment`,
    price: 50,
    raiting: 3,
    type: `Apartment`,
    isPremium: false,
    bedrooms: 3,
    guests: 2,
    householdItems: new Set([
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
    ]),
    host: {
      name: `Angelina`,
      isPro: true,
      avatarUrl: `img/avatar-angelina.jpg`,
    },
    location: [52.379553, 4.873096]
  },
];

export {offers};
