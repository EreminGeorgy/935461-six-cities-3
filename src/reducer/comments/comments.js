import {extend} from '../../utils/utils.js';
import {ApplicationApi} from "../../application-api.js";

const CommentsOperationStatus = {
  SUCCESS: `SUCCESS`,
  ERROR: `ERROR`,
  REQUEST: `REQUEST`,
};

const CommentsActions = {
  SUCCESS: `COMMENTS_SUCCESS`,
  REQUEST: `COMMENTS_REQUEST`,
  FAILURE: `COMMENTS_FAILURE`,
};

const initialState = {
  commentsOperationStatus: CommentsOperationStatus.SUCCESS,
};

const ActionCreator = {
  success: () => {
    return {
      type: CommentsActions.SUCCESS,
      payload: CommentsOperationStatus.SUCCESS,
    };
  },
  request: () => {
    return {
      type: CommentsActions.REQUEST,
      payload: CommentsOperationStatus.REQUEST,
    };
  },
  failure: () => {
    return {
      type: CommentsActions.FAILURE,
      payload: CommentsOperationStatus.ERROR,
    };
  },
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case CommentsActions.SUCCESS:
      return extend(state, {
        commentsOperationStatus: action.payload,
      });
    case CommentsActions.REQUEST:
      return extend(state, {
        commentsOperationStatus: action.payload,
      });
    case CommentsActions.FAILURE:
      return extend(state, {
        commentsOperationStatus: action.payload,
      });
  }

  return state;
};
/*eslint-disable */
const Operation = {
  sendComment: (commentData) => (dispatch, getState, api) => {
    dispatch(ActionCreator.request());
    return ApplicationApi.sendComment(commentData)
      .then((response) => {
        dispatch(ActionCreator.success());
      })
      .catch((err) => {
        dispatch(ActionCreator.failure());
        const {response} = err;

        // if (response && response.status === UNAUTHORIZED) {
        //   dispatch(UserActionCreator.requireAuthorization(AuthorizationStatus.NO_AUTH))
        //   window.location.pathname = AppRoute.AUTH;
        // }
        throw err;
      });
  },
  /*eslint-disable */
};


export {
  ActionCreator,
  CommentsActions,
  CommentsOperationStatus,
  Operation,
  reducer,
};
