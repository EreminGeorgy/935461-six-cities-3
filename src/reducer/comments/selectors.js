import NameSpace from "../name-space";

const NAME_SPACE = NameSpace.COMMENTS;

export const getCommentState = (state) => {
  return state[NAME_SPACE].commentsOperationStatus;
};

export const getComments = (state) => {
  return state[NAME_SPACE].comments;
};

