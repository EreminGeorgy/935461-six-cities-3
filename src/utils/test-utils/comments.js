
export const COMMENTS = [{
  comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  dateString: new Date(`2019-05-08T14:13:56.569Z`),
  id: 1,
  rating: 4,
  user: {
    avatarUrl: `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/13.jpg`,
    id: 4,
    isPro: false,
    name: `Max`
  }
}];
