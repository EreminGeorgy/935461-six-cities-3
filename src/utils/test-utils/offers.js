const shortid = require(`shortid`);

export const offers = [
  {
    id: shortid.generate(),
    previewSrc: `img/apartment-01.jpg`,
    title: `_`,
    price: 0,
    raiting: 0,
    type: `Apartment`,
    isPremium: true,
  },
  {
    id: shortid.generate(),
    previewSrc: `img/apartment-02.jpg`,
    title: `Wood and stone place`,
    price: 90,
    raiting: 8.35,
    type: `Private room`,
    isPremium: false,
  },
  {
    id: shortid.generate(),
    previewSrc: `img/apartment-03.jpg`,
    title: `Canal View Prinsengracht`,
    price: 80,
    raiting: 8.2,
    type: `1`,
    isPremium: true,
  },
  {
    id: shortid.generate(),
    previewSrc: `img/room.jpg`,
    title: `Nice, cozy, warm big bed apartment`,
    price: 50,
    raiting: 3,
    type: `Apartment`,
    isPremium: false,
  },
];
