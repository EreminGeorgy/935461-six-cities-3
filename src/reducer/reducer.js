import {offers} from '../mocks/offers';
import {extend, getCities, getOffersByCity} from '../utils/utils.js';

const citiesList = getCities(offers);
const initialOffers = getOffersByCity(offers, citiesList[0]);

const initialState = {
  activeCity: citiesList[0],
  offersInActiveCity: initialOffers,
  cities: citiesList,
};

const ActionCreator = {
  newCity: (newCity) => ({
    type: `NEW_CITY`,
    payload: newCity,
  }),
  getOffers: (newCity) => ({
    type: `GET_OFFERS`,
    payload: newCity,
  }),
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case `NEW_CITY`:
      return extend(state, {
        activeCity: action.payload,
      });
    case `GET_OFFERS`:
      return extend(state, {
        offersInActiveCity: getOffersByCity(offers, action.payload),
      });
  }
  return state;
};

export {reducer, ActionCreator};
