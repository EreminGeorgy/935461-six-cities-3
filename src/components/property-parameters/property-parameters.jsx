import React from "react";
import PropTypes from "prop-types";
import {getStars} from "../../utils/utils.js";
import {Review} from "../review/review.jsx";
import {COMMENTS} from "../../utils/test-utils/comments.js";

export const PropertyParameters = (props) => {
  const {offer} = props;
  const {
    imagesSrc,
    title,
    description,
    price,
    rating,
    type,
    isPremium,
    bedrooms,
    guests,
    householdItems,
    host,
  } = offer;

  const width = getStars(rating);
  const availableItems = Array.from(householdItems);

  return (
    <section className="property">
      <div className="property__gallery-container container">
        <div className="property__gallery">
          {imagesSrc.map((src, i) => {
            return (<div className="property__image-wrapper" key={`src-${i}`}>
              <img className="property__image" src={src} alt="Photo studio"/>
            </div>);
          })}
        </div>
      </div>
      <div className="property__container container">
        <div className="property__wrapper">
          {isPremium ? (<div className="property__mark"><span>Premium</span></div>) : (``)}
          <div className="property__name-wrapper">
            <h1 className="property__name">{title}</h1>
            <button className="property__bookmark-button button" type="button">
              <svg className="property__bookmark-icon" width="31" height="33">
                <use xlinkHref="#icon-bookmark"></use>
              </svg>
              <span className="visually-hidden">To bookmarks</span>
            </button>
          </div>
          <div className="property__rating rating">
            <div className="property__stars rating__stars">
              <span style={{width}}></span>
              <span className="visually-hidden">Rating</span>
            </div>
            <span className="property__rating-value rating__value">{rating}</span>
          </div>
          <ul className="property__features">
            <li className="property__feature property__feature--entire">
              {type}
            </li>
            <li className="property__feature property__feature--bedrooms">
              {bedrooms} Bedrooms
            </li>
            <li className="property__feature property__feature--adults">
              Max {guests} adults
            </li>
          </ul>
          <div className="property__price">
            <b className="property__price-value">&euro;{price}</b>
            <span className="property__price-text">&nbsp;night</span>
          </div>
          <div className="property__inside">
            <h2 className="property__inside-title">What&apos;s inside</h2>
            <ul className="property__inside-list">
              {availableItems.map((item) => {
                return (<li className="property__inside-item" key={item.toString()}>
                  {item}
                </li>);
              })}
            </ul>
          </div>April 2019
          <div className="property__host">
            <h2 className="property__host-title">Meet the host</h2>
            <div className="property__host-user user">
              <div className={`property__avatar-wrapper ${host.isPro ? `property__avatar-wrapper--pro` : ``} user__avatar-wrapper`}>
                <img className="property__avatar user__avatar" src={host.avatarUrl} width="74" height="74" alt="Host avatar"/>
              </div>
              <span className="property__user-name">
                {host.name}
              </span>
            </div>
            <div className="property__description">
              <p className="property__text" >{description}</p>;
            </div>
          </div>
          <section className="property__reviews reviews">
            <h2 className="reviews__title">Reviews &middot; <span className="reviews__amount">1</span></h2>
            <ul className="reviews__list">
              {COMMENTS.map((commentData) => {
                return <Review
                  key={commentData.id}
                  commentData={commentData}
                />;
              })}
            </ul>
            <form className="reviews__form form" action="#" method="post">
              <label className="reviews__label form__label" htmlFor="review">Your review</label>
              <div className="reviews__rating-form form__rating">
                <input className="form__rating-input visually-hidden" name="rating" value="5" id="5-stars" type="radio" />
                <label htmlFor="5-stars" className="reviews__rating-label form__rating-label" title="perfect">
                  <svg className="form__star-image" width="37" height="33">
                    <use xlinkHref="#icon-star"></use>
                  </svg>
                </label>

                <input className="form__rating-input visually-hidden" name="rating" value="4" id="4-stars" type="radio" />
                <label htmlFor="4-stars" className="reviews__rating-label form__rating-label" title="good">
                  <svg className="form__star-image" width="37" height="33">
                    <use xlinkHref="#icon-star"></use>
                  </svg>
                </label>

                <input className="form__rating-input visually-hidden" name="rating" value="3" id="3-stars" type="radio" />
                <label htmlFor="3-stars" className="reviews__rating-label form__rating-label" title="not bad">
                  <svg className="form__star-image" width="37" height="33">
                    <use xlinkHref="#icon-star"></use>
                  </svg>
                </label>

                <input className="form__rating-input visually-hidden" name="rating" value="2" id="2-stars" type="radio" />
                <label htmlFor="2-stars" className="reviews__rating-label form__rating-label" title="badly">
                  <svg className="form__star-image" width="37" height="33">
                    <use xlinkHref="#icon-star"></use>
                  </svg>
                </label>

                <input className="form__rating-input visually-hidden" name="rating" value="1" id="1-star" type="radio" />
                <label htmlFor="1-star" className="reviews__rating-label form__rating-label" title="terribly">
                  <svg className="form__star-image" width="37" height="33">
                    <use xlinkHref="#icon-star"></use>
                  </svg>
                </label>
              </div>
              <textarea className="reviews__textarea form__textarea" id="review" name="review" placeholder="Tell how was your stay, what you like and what can be improved"></textarea>
              <div className="reviews__button-wrapper">
                <p className="reviews__help">
                  To submit review please make sure to set <span className="reviews__star">rating</span> and describe your stay with at least <b className="reviews__text-amount">50 characters</b>.
                </p>
                <button className="reviews__submit form__submit button" type="submit" disabled="">Submit</button>
              </div>
            </form>
          </section>
        </div>
      </div>
      <section className="property__map map"></section>
    </section>
  );
};

PropertyParameters.propTypes = {
  offer: PropTypes.shape({
    imagesSrc: PropTypes.arrayOf(PropTypes.string).isRequired,
    title: PropTypes.string.isRequired,
    description: PropTypes.string,
    price: PropTypes.number.isRequired,
    rating: PropTypes.number.isRequired,
    type: PropTypes.string,
    isPremium: PropTypes.bool,
    bedrooms: PropTypes.number.isRequired,
    guests: PropTypes.number.isRequired,
    householdItems: PropTypes.objectOf(PropTypes.string),
    host: PropTypes.shape({
      name: PropTypes.string.isRequired,
      avatarUrl: PropTypes.string.isRequired,
      isPro: PropTypes.bool
    })
  }).isRequired,
};
