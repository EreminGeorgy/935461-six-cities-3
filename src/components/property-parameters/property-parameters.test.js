import React from "react";
import renderer from "react-test-renderer";
import {PropertyParameters} from "./property-parameters.jsx";
import {offer, offers} from "../../utils/test-utils/offers.js";
import {Provider} from 'react-redux';
import reducer from '../../reducer/reducer.js';
import {createStore} from 'redux';
import {COMMENTS} from "../../utils/test-utils/comments.js";

const store = createStore(
    reducer
);

const closestOffers = offers;

it(`<PropertyParameters /> should render PropertyParameters`, () => {
  const tree = renderer
    .create(<Provider store={store}>
      <PropertyParameters
        offer={offer}
        closestOffers={closestOffers}
        activeCard={1}
        comments={COMMENTS}
      />
    </Provider>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
