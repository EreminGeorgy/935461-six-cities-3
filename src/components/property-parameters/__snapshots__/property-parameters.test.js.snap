// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PropertyParameters /> should render PropertyParameters 1`] = `
<section
  className="property"
>
  <div
    className="property__gallery-container container"
  >
    <div
      className="property__gallery"
    >
      <div
        className="property__image-wrapper"
      >
        <img
          alt="Photo studio"
          className="property__image"
          src="img/apartment-01.jpg"
        />
      </div>
      <div
        className="property__image-wrapper"
      >
        <img
          alt="Photo studio"
          className="property__image"
          src="img/apartment-02.jpg"
        />
      </div>
      <div
        className="property__image-wrapper"
      >
        <img
          alt="Photo studio"
          className="property__image"
          src="img/apartment-01.jpg"
        />
      </div>
      <div
        className="property__image-wrapper"
      >
        <img
          alt="Photo studio"
          className="property__image"
          src="img/apartment-03.jpg"
        />
      </div>
      <div
        className="property__image-wrapper"
      >
        <img
          alt="Photo studio"
          className="property__image"
          src="img/apartment-02.jpg"
        />
      </div>
      <div
        className="property__image-wrapper"
      >
        <img
          alt="Photo studio"
          className="property__image"
          src="img/room.jpg"
        />
      </div>
    </div>
  </div>
  <div
    className="property__container container"
  >
    <div
      className="property__wrapper"
    >
      <div
        className="property__mark"
      >
        <span>
          Premium
        </span>
      </div>
      <div
        className="property__name-wrapper"
      >
        <h1
          className="property__name"
        >
          title
        </h1>
        <button
          className="property__bookmark-button button "
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="33"
            width="31"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="property__rating rating"
      >
        <div
          className="property__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "200%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
        <span
          className="property__rating-value rating__value"
        >
          10
        </span>
      </div>
      <ul
        className="property__features"
      >
        <li
          className="property__feature property__feature--entire"
        >
          Apartment
        </li>
        <li
          className="property__feature property__feature--bedrooms"
        >
          3
           Bedrooms
        </li>
        <li
          className="property__feature property__feature--adults"
        >
          Max 
          2
           adults
        </li>
      </ul>
      <div
        className="property__price"
      >
        <b
          className="property__price-value"
        >
          €
          0
        </b>
        <span
          className="property__price-text"
        >
           night
        </span>
      </div>
      <div
        className="property__inside"
      >
        <h2
          className="property__inside-title"
        >
          What's inside
        </h2>
        <ul
          className="property__inside-list"
        >
          <li
            className="property__inside-item"
          >
            Wi-Fi
          </li>
          <li
            className="property__inside-item"
          >
            Washing machine
          </li>
          <li
            className="property__inside-item"
          >
            Towels
          </li>
        </ul>
      </div>
      April 2019
      <div
        className="property__host"
      >
        <h2
          className="property__host-title"
        >
          Meet the host
        </h2>
        <div
          className="property__host-user user"
        >
          <div
            className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper"
          >
            <img
              alt="Host avatar"
              className="property__avatar user__avatar"
              height="74"
              src="img/avatar-angelina.jpg"
              width="74"
            />
          </div>
          <span
            className="property__user-name"
          >
            Angelina
          </span>
        </div>
        <div
          className="property__description"
        >
          <p
            className="property__text"
          >
            A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.
          </p>
        </div>
      </div>
      <section
        className="property__reviews reviews"
      >
        <h2
          className="reviews__title"
        >
          Reviews · 
          <span
            className="reviews__amount"
          >
            1
          </span>
        </h2>
        <ul
          className="reviews__list"
        >
          <li
            className="reviews__item"
          >
            <div
              className="reviews__user user"
            >
              <div
                className="reviews__avatar-wrapper user__avatar-wrapper"
              >
                <img
                  alt="Reviews avatar"
                  className="reviews__avatar user__avatar"
                  height="54"
                  src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/13.jpg"
                  width="54"
                />
              </div>
              <span
                className="reviews__user-name"
              >
                Max
              </span>
            </div>
            <div
              className="reviews__info"
            >
              <div
                className="reviews__rating rating"
              >
                <div
                  className="reviews__stars rating__stars"
                >
                  <span
                    style={
                      Object {
                        "width": "80%",
                      }
                    }
                  />
                  <span
                    className="visually-hidden"
                  >
                    Rating
                  </span>
                </div>
              </div>
              <p
                className="reviews__text"
              >
                A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
              </p>
              <time
                className="reviews__time"
                dateTime="2019-05-08"
              >
                May 2019
              </time>
            </div>
          </li>
        </ul>
        
      </section>
    </div>
  </div>
  <section
    className="cities__map"
  >
    <div
      id="mapId"
      style={
        Object {
          "height": "580px",
        }
      }
    />
  </section>
</section>
`;
